tox.ini

```diff
@@ -1,62 +1,81 @@
+# This file is centrally managed as a template file in https://github.com/canonical/solutions-engineering-automation
+# To update the file:
+# - Edit it in the canonical/solutions-engineering-automation repository.
+# - Open a PR with the changes.
+# - When the PR merges, the soleng-terraform bot will open a PR to the target repositories with the changes.
+
 [tox]
 skipsdist=True
 skip_missing_interpreters = True
-envlist = lint, unit, func
+envlist = lint, unit
 
 [testenv]
 basepython = python3
 setenv =
-  PYTHONPATH = {toxinidir}:{toxinidir}/lib/:{toxinidir}/hooks/
-passenv =
-  CHARM_*
-  HOME
-  PATH
-  PYTEST_KEEP_MODEL
-  PYTEST_CLOUD_NAME
-  PYTEST_CLOUD_REGION
-  PYTEST_MODEL
-  MODEL_SETTINGS
-  HTTP_PROXY
-  HTTPS_PROXY
-  NO_PROXY
-  SNAP_HTTP_PROXY
-  SNAP_HTTPS_PROXY
-  OS_*
-  TEST_*
+  PYTHONPATH = {toxinidir}:{toxinidir}/src/:{toxinidir}/reactive/:{toxinidir}/hooks/:{toxinidir}/lib/:{toxinidir}/actions:{toxinidir}/files/:{toxinidir}/files/plugins/
+  # avoid state written to file during tests - see https://github.com/juju/charm-helpers/blob/85dcbeaf63b0d0f38e8cb17825985460dc2cd02d/charmhelpers/core/unitdata.py#L179-L184
+  UNIT_STATE_DB = :memory:
+passenv = *
 
 [testenv:lint]
 commands =
+    black --check --diff --color .
+    isort --check --diff --color .
     flake8
-    black --check .
+    pylint --recursive=y .
+    # mypy temporarily disabled because it doesn't gracefully recursively find files like the others
+    #mypy --install-types --non-interactive .
 deps =
     black
+    colorama
     flake8
-    flake8-pyproject
+    flake8-colors
     flake8-docstrings
     flake8-import-order
+    flake8-pyproject
+    isort
+    #mypy
     pep8-naming
-    flake8-colors
+    pylint
+    # so pylint and mypy can reason about the code
+    {[testenv:unit]deps}
+    {[testenv:func]deps}
 
 [testenv:reformat]
 commands =
     black .
+    isort .
 deps =
     black
+    isort
 
 [testenv:unit]
-commands =
-    coverage run -m unittest discover -s {toxinidir}/tests/unit -t {toxinidir} -v
-    coverage report --omit tests/*,mod/*,.tox/*
-    coverage html --omit tests/*,mod/*,.tox/*
-deps = -r{toxinidir}/tests/unit/requirements.txt
 setenv =
     {[testenv]setenv}
     COVERAGE_FILE = .coverage-unit
+commands = pytest {toxinidir}/tests/unit \
+   -v \
+   --cov \
+   --cov-report=term-missing \
+   --cov-report=html \
+   --cov-report=xml \
+   {posargs}
+
```

Comment:
project use zaza. Need to change in automation tool

Reply:
I'm confused - this is the unit tests, which have been updated from `coverage run -m unittest ...` to pytest. Zaza is still being used for the func tests (see below).

Reply:
Sorry, I think I got a little confused about adding `pytest` for the functional tests and commented in the wrong place, but everything seems to be good